0162-semantic-convention-resource-stability.md

```diff
@@ -0,0 +1,110 @@
+# Resource Semantic Convention Stabiltiy Requirements
+
+Outline what changes are "safe" to perform for resource semantic conventions
+and which require major version bumps.
+
+## Motivation
+
+This outlines acceptable changes to semantic conventions for resources, allowing
+growth and evolution over time.
+
+While Telemetry Schemas propose a mechanism to define changes to semantic
+conventions in a way that users can evolve telemetry, this document first
+attempts to define what is considered a compatible change.
+
+## Explanation
+
+We take the assumption that `Resource`, as it exists today, is the identity of
+an SDK and is leveraged as such in a lot of assumptions throughout open
+telemetry. Specifically, if an SDK that used to produce a `Resource` of
+attributes XYZ instead produces ABC, these metrics would be considered from a
+different source. To that end, if a Metric produced on version x.y of
+OpenTelemetry SDK is incompatible with one produced on version x.{y+1} because
+of `Resource` changes, this would be considered breaking.
+
+This means the following:
+
+- `Resource` detectors should provide a comprehensive set of attributes in
+  all or nothing fashion. if a resource detector produced attributes XYZ for
+  a given compute, it should continue to do so for compatible versions.
+- Resource attributes (e.g. `cloud.provider`) may have new possible values
+  defined as long as there is zero overlap with how the previous values were
+  computed.  For example, `acme-cloud` would never be inferred where previously
+  `aws` was inferred.
+  *Note: if `aws` was inferred incorrectly on "Acme cloud" previously, that would not be a violation.*
+- Adding new Resource attribute semantic conventions is allowed, however
+  existing resource detectors MUST NOT use them, only new detectors would be
+  allowed.
```

Comment:
This last point is a bit vague. What counts as a new detector? If an attribute is added to the K8s resource definition, how would it be handled?